sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamCalcRel.java

```diff
@@ -217,10 +226,14 @@ public Calc copy(RelTraitSet traitSet, RelNode input, RexProgram program) {
                       Types.lookupMethod(DoFn.ProcessContext.class, "output", Object.class),
                       output))));
 
-      CalcFn calcFn = new CalcFn(builder.toBlock().toString(), outputSchema);
+      CalcFn calcFn = new CalcFn(builder.toBlock().toString(), outputSchema, getJarPaths(program));
 
       // validate generated code
-      calcFn.compile();
+      try {
+        calcFn.compile();
+      } catch (IOException e) {
```

Comment:
nit: possibly handle this inside `compile()` so it doesn't expose an IOException?

Reply:
done